### ISSUE TITLE
Question about schema correctness 

### ISSUE THREAD

**Issue by sungam3r** [graphql/graphql-spec#653]
Is this schema correct? Inadvertently, I redefined the type.

```graphql
type Query {
  SendEmail(messageDto: EmailMessageDtoInput): Boolean
}

input EmailMessageDtoInput {
  url: String
  data: String 
}

input EmailMessageDtoInput {
  url: String
  data: String 
}
```
With 99% probability, I guess the answer is NO. However, does the specification speak explicitly about this? I spent a lot of time to find the source of error in one of out services running graphql-dotnet package. It turned out that the error was due to the fact that the type was re-defined in the service SDL. Now I think where it is worth doing the type check - at the [graphql-dotnet](https://github.com/graphql-dotnet) level or at the [graphql-parser](https://github.com/graphql-dotnet/parser) (lower) level - graphql-dotnet uses parsed SDL document from graphql-parser. In general, I am inclined to do a check in both places. I just want to get confirmation that this situation is really unacceptable.

**Comment by sungam3r** [comment]
I think I already found the answer: https://graphql.github.io/graphql-spec/June2018/#sec-Schema
> All types within a GraphQL schema must have unique names. No two provided types may have the same name. No provided type may have a name which conflicts with any built in types (including Scalar and Introspection types).

**Comment by sungam3r** [comment]
But now I will again ask a question about SDL validation: https://github.com/graphql/graphql-spec/issues/648#issuecomment-557544759

This is the second time that there is a need to formalize the requirements for SDL correctness.

**Comment by martinbonnin** [comment]
> All types within a GraphQL schema must have unique names.

Yup, this is it! I'm going to go ahead and assumed you found your answer in the last 7 years ^^! Thanks for opening this!
